### PR TITLE
APPSEC-42 fix csrf vulnerability

### PR DESCRIPTION
**Pull request by nathanDenny1**
Fixes a CSRF vulnerability for old versions of Wordpress

### CODE REVIEW

File: setup.php
Comment (nathanDenny1) [inline setup.php:1]: This is just setting then checking for a nonce on form submit. This prevents someone from submitting to this endpoint from a different source. WordPress nonce documentation: https://developer.wordpress.org/apis/security/nonces/

File: setup.php [hunk 1]
```diff
@@ -1,5 +1,10 @@
 <?php
-if (!empty($_POST['checkfront_host'])) {
+if (!empty($_POST['checkfront_host']) && !empty($_POST['_wpnonce'])) {
+	//prevent CSRF attacks
+	if (!wp_verify_nonce($_POST['_wpnonce'], 'update_checkfront_host')) {
```
Comment (david-light) [inline setup.php:4]: Should we be logging anything here?

Comment (nathanDenny1) [reply setup.php:4]: Yes, probably. However, we don't have logging set up for the Wordpress Plugin, so the logs wouldn't go anywhere. We could ask DevOps to setup logging for this repo, but I'm not sure how worth it that would be

**Review by connkat — APPROVED** [review]
LGTM

**Review by david-light — APPROVED** [review]

**Pull request merged**